### PR TITLE
Add `disabled` prop to `SortableContext`

### PR DESCRIPTION
**Pull request by otofu-square**
close https://github.com/clauderic/dnd-kit/issues/618

### CODE REVIEW

File: packages/sortable/src/hooks/useSortable.ts [hunk 1]
```diff
@@ -64,13 +66,18 @@ export function useSortable({
     () => items.slice(items.indexOf(id)),
     [items, id]
   );
+  const droppableDisabled =
+    typeof disabled === 'boolean' ? false : !!disabled?.droppable;
```
Comment (otofu-square) [inline packages/sortable/src/hooks/useSortable.ts:70]: `droppableDisabled` is always false to ensure backward compatibility.

File: packages/sortable/src/components/SortableContext.tsx [hunk 2]
```diff
@@ -94,6 +98,7 @@ export function SortableContext({
       useDragOverlay,
       sortedRects: getSortedRects(items, droppableRects),
       strategy,
+      disabled,
```
Comment (otofu-square) [inline packages/sortable/src/components/SortableContext.tsx:101]: Passing disabled to SortableContext makes it possible to disable all items in the context

File: packages/sortable/src/hooks/useSortable.ts [hunk 3]
```diff
@@ -64,13 +66,18 @@ export function useSortable({
...
+  const draggableDisabled =
+    typeof disabled === 'boolean' ? disabled : !!disabled?.draggable;
   const {rect, node, isOver, setNodeRef: setDroppableNodeRef} = useDroppable({
     id,
     data,
     resizeObserverConfig: {
       updateMeasurementsFor: itemsAfterCurrentSortable,
       ...resizeObserverConfig,
     },
+    disabled: globalDisabled || droppableDisabled,
```
Comment (otofu-square) [inline packages/sortable/src/hooks/useSortable.ts:80]: Evaluate the value of disabled in SortableContext first.

File: packages/sortable/src/components/SortableContext.tsx [hunk 4]
```diff
@@ -11,6 +11,7 @@ export interface Props {
   items: (UniqueIdentifier | {id: UniqueIdentifier})[];
   strategy?: SortingStrategy;
   id?: string;
+  disabled?: boolean | { draggable?: boolean; droppable?: boolean }; };
```
Comment (otofu-square) [inline packages/sortable/src/components/SortableContext.tsx:14]: Passing `disabled` to `SortableContext` makes it possible to disable all items in the context

File: packages/sortable/src/hooks/useSortable.ts [hunk 5]
```diff
@@ -64,13 +67,20 @@ export function useSortable({
     () => items.slice(items.indexOf(id)),
     [items, id]
   );
+  const droppableDisabled =
+    globalDroppableDisabled ||
+    (typeof disabled === 'boolean' ? false : !!disabled?.droppable);
```
Comment (otofu-square) [inline packages/sortable/src/hooks/useSortable.ts:72]: `droppableDisabled` is always false to ensure backward compatibility.

**Comment by clauderic** [comment]
Closing in favour of https://github.com/clauderic/dnd-kit/pull/754

**Pull request closed without merging**